asciidoctorj-core/src/main/java/org/asciidoctor/ast/AbstractBlockImpl.java

```diff
@@ -81,4 +90,46 @@ public int getLevel() {
         return new RubyBlockListDecorator(rubyBlocks);
     }
 
+    @Override
+    public void parse(List<String> lines) {
+        Parser parser = new Parser(lines);
+
+        AbstractBlock nextBlock = parser.nextBlock();
+        while (nextBlock != null) {
+            append(nextBlock);
+            nextBlock = parser.nextBlock();
+        }
+    }
+
+    private class Parser extends RubyObjectWrapper {
+
+        private final IRubyObject reader;
+
+        public Parser(List<String> lines) {
+            super(AbstractBlockImpl.this.getRuntime().getModule("Asciidoctor").getClass("Parser"));
+
+            RubyArray rubyLines = runtime.newArray(lines.size());
+            for (String line: lines) {
+                rubyLines.add(runtime.newString(line));
+            }
+
+            RubyClass readerClass = runtime.getModule("Asciidoctor").getClass("Reader");
+            this.reader = readerClass.callMethod("new", rubyLines);
+        }
+
+        public AbstractBlock nextBlock() {
+            try {
+                IRubyObject nextBlock = getRubyProperty("next_block", reader, AbstractBlockImpl.this.getRubyObject());
+                if (nextBlock.isNil()) {
+                    return null;
+                } else {
+                    return (AbstractBlock) NodeConverter.createASTNode(nextBlock);
+                }
+            } catch (Exception e) {
+                e.printStackTrace();
```

Comment:
Oops, that certainly must be removed.
Maybe we could add checkstyle to the build so that things like that don't happen.